docs/ldml/tr35-keyboards.md

```diff
@@ -1348,206 +1343,503 @@ This example shows some of the mappings for a French keyboard layout:
+- **Nested non-capture groups**
 
-For example:
+    `(?:thismatches) (?:[abc]([def]))|(?:[ghi])`
 
-```xml
-<transform from="\u037A\u037A" to="\u037A" error="fail" />
-```
+    Only the innermost group is allowed to be a capture group.
 
-This indicates that it is an error to type two iota subscripts immediately after each other.
+- **Disjunctions**
 
-In terms of how these different attributes work in processing a sequence of transforms, consider the transform:
+    `abc|def`
 
-```xml
-<transform before="X" from="Y" to="B" />
-```
+    Match either `abc` or `def`.
 
-This would transform the string:
+- **Match a single Unicode codepoint**
 
-```default
-XYZ → XBZ
-```
+    `.`
 
-If we mark where the current match position is before and after the transform we see:
+    Matches a codepoint, not individual code units. (See the ’u’ option in EcmaScript262 regex.)
+    For example, Osage `𐒵` is one match (`.`) not two.
+    Does not match [markers](#markers). (See `\m{.}` and `\m{marker}`, below.)
 
-```default
-X | Y Z → X B | Z
-```
+- **Match the start of the text context**
 
-And a subsequent transform could transform the Z string, looking back (using @before) to match the B.
+    `^`
 
-* * *
+    The start of the context could be the start of a line, a grid cell, or some other formatting boundary.
+    See description at the top of [`transforms`](#element-transform).
 
-### <a name="Element_reorders" href="#Element_reorders">Element: reorders</a>
+#### Additional Features
 
-This element defines a group of one or more `reorder` elements associated with this keyboard layout.
+The following are additions to standard Regex syntax.
 
-> <small>
->
-> Parents: [keyboard](#Element_keyboard)
+- **Match a Marker**
+
+    `\m{Some_Marker}`
+
+    Matches the named marker.
+    Also see [Markers](#markers).
+
+- **Match a single marker**
+
+    `\m{.}`
+
+    Matches any single marker.
+    Also see [Markers](#markers).
+
+- **Variables**
+
+    `${zwnj}`
+
+    In this usage, the variable with `id="Matras"` will be substituted in at this point in the expression. The variable can contain a range, a character, or any other portion of a pattern. If `zwnj` is a simple string, the pattern will match that string at this point.
+
+- **Set variables**
+
+    `$[upper]`
+
+    Given a space-separated variable, this syntax will match _any_ of the substrings. This expression may be thought of  (and implemented) as if it were a _non-capturing group_. It may, however, be enclosed within a capturing group.
+
+    Given the variable `upper` defined below, the expression `$[upper]` will match as if it were written `(?:A|B|CC|D|E|FF)`. Note that leading, trailing, and excess whitespace are ignored.
+
+    ```xml
+    <variable id="upper" value=" A B CC  D E  FF " />
```

Comment:
Is there a reason for the extra spaces after certain entries?

```suggestion
    <variable id="upper" value=" A B CC D E FF " />
```

Reply:
yes, it shows that they can be aligned with the prior and succeeding lines (in this case 'upper' and 'lower' are a pair)

Reply:
That certainly holds in a later example I saw during the review, but in isolation it felt like the extra spaces may have been implying something I wasn't inferring.  It's certainly clearer later on once the alignment example comes into play.

Reply:
Think this is now covered, reopen if not

Reply:
LGTM